joint_state_broadcaster/include/joint_state_broadcaster/joint_state_broadcaster.hpp

```diff
@@ -94,14 +95,16 @@ class JointStateBroadcaster : public controller_interface::ControllerInterface
   //  we store the name of joints with compatible interfaces
   std::vector<std::string> joint_names_;
   std::shared_ptr<rclcpp::Publisher<sensor_msgs::msg::JointState>> joint_state_publisher_;
-  sensor_msgs::msg::JointState joint_state_msg_;
+  std::shared_ptr<realtime_tools::RealtimePublisher<sensor_msgs::msg::JointState>>
+    realtime_joint_state_publisher_ = nullptr;
```

Comment:
Not sure about 'realtime' prefix though...

I would call the normal one with 'pub_' suffix and this one 'publisher_'


One way or another we should sync naming in all broadcasters we have.

Reply:
I didn't have a better idea... I actually want to discuss the realtime publisher API, I'm r ally.not happy with it